### ISSUE TITLE
CarouselPageChangedReason not working

### ISSUE THREAD

**Issue by dannyalbuquerque** [serenader2014/flutter_carousel_slider#133]
if i use this in the "Manually controlled slider" to display the onPageChanged reason. It's always `CarouselPageChangedReason.controller`. 

`onPageChanged: (index, reason) => print(reason.toString()))`

result:
```
I/flutter ( 3892): CarouselPageChangedReason.controller
I/flutter ( 3892): CarouselPageChangedReason.controller
I/flutter ( 3892): CarouselPageChangedReason.controller
```

`CarouselPageChangedReason.controller` should only appear when calling `_controller.animateToPage` and `CarouselPageChangedReason.manual` when swiping by hand.

**Comment by atrope** [comment]
@dannyalbuquerque feel free to use and report bugs in my branch:
#136 

```
  carousel_slider:
    git:
      url: https://github.com/SuaMusica/flutter_carousel_slider.git
      ref: fix/reasonPageChange

```

**Comment by serenader2014** [comment]
Fixed in v2.1.0
